### ISSUE TITLE
Sync Incomplete: Incorrect "...not associated with a contact." error message

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#4031]
**[Mingle Card: 4363](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/4363)**
Steps to Reproduce
------------------

```
| | |
|-|-|
|**Version #**|4.0.1.3342|
|**Hardware**|Dell Latitude E6530|
|**OS**|Windows 10 Pro|
|**Browser**|Chrome 52.0.2743.116 m|
|**Username**|mattfield@lf40.com|
|**Password**|computer.45|
  
 

```

1.  Log in to **FieldFX Mobile** with the **Matt Field** account.
2.  The initial sync fails with the error message, **This user is not associated with a contact**.
3.  Log in to **FieldFX Back Office**, click the **Contacts** tab, and select the **Matt Field** contact.
4.  As you can see, the Matt Field contact is linked to the Matt Field user.  See the attached **Sync\_Incomplete.png** for more info.

Expected Result
---------------

Either allow the sync to complete when the contact does not have an account or if the account really is necessary then we need to print a clear message that your contact does not have an account.

Actual Result
-------------

I can’t sync because the Contact isn’t associated an Account.  If you go back to the Contact and add an Account to the user, you’ll now complete the sync.

Note: In order to sync you must have a User → Contact → Account.

**TL;DR**: The error message needs to be updated to tell the user about adding the contact to an account.

Analysis
--------

Not able to reproduce the issue on tarantula 3. But worked with matt and was able to reproduce on roy’s QA org(below) and matt’s test4 org.  When the account field is removed from the contact, the user associated with the contact no longer has access to the contact itself. The error message on mobile is accurate as the user no longer has a contact. On console, I couldn’t find anything that created this relationship, but roy pointed me to an article that appears to indicate a salesforce bug around this issue. 

[https://help.salesforce.com/apex/HTViewSolution?id=000004005&language=en\_US](https://www.google.com/url?q=https%3A%2F%2Fhelp.salesforce.com%2Fapex%2FHTViewSolution%3Fid%3D000004005%26language%3Den_US&sa=D&sntz=1&usg=AFQjCNFQ33KqhUd2XVFw2ZKDeNDqf5qaCw)

Related Cards
-------------

Test Plan
---------

TBD
